assets/js/pluginsidebar/components/notifications/index.js

```diff
@@ -0,0 +1,216 @@
+      return;
+    }
+
+    // Loop over the array of notifications and determine which ones we need to clear.
+    const toClear = notifications
+      .filter((notification) => true !== notification.dismissible);
+
+    // Ensure there are items to be cleared.
+    if (0 === toClear.length) {
+      return;
+    }
+
+    // Send the request to the API to clear the notifications.
+    apiFetch({
+      data: {
+        toClear,
+      },
+      method: 'POST',
+      path: '/apple-news/v1/clear-notifications',
+    })
+      .catch((error) => console.error(error)); /* eslint-disable-line no-console */
+  }
+
+  /**
+   * A callback for a dismiss action on a notification.
+   * @param {object} notification - The notification to mark as dismissed.
+   */
+  dismissNotification(notification) {
+    const {
+      apiFetch,
+    } = wp;
+    const {
+      notifications,
+    } = this.state;
+
+    // Send the request to the API to clear the notification.
+    apiFetch({
+      data: {
+        toClear: [notification],
+      },
+      method: 'POST',
+      path: '/apple-news/v1/clear-notifications',
+    })
+      .then(() => {
+        // Set the notification to dismissed and update state.
+        const updatedNotifications = notifications.map((compare) => {
+          // If the notification doesn't match, return as-is.
+          if (JSON.stringify(compare) !== JSON.stringify(notification)) {
+            return compare;
+          }
+
+          return {
+            ...compare,
+            dismissed: true,
+          };
+        });
+        this.setState({
+          notifications: updatedNotifications,
+        });
+      })
+      .catch((error) => console.error(error)); /* eslint-disable-line no-console */
+  }
+
+  /**
+   * Fetches notifications for the current user via the REST API.
+   */
+  fetchNotifications() {
+    const {
+      apiFetch,
+    } = wp;
+    const path = '/apple-news/v1/get-notifications';
+    apiFetch({ path })
+      .then((notifications) => {
+        if (Array.isArray(notifications)) {
+          if (0 < notifications.length) {
+            this.setState(
+              {
+                notifications,
+              },
+              this.clearNotifications
+            );
+          } else {
+            this.setState({
+              notifications,
+            });
+          }
+        }
+      })
+      .catch((error) => console.error(error)); /* eslint-disable-line no-console */
+  }
+
+  /**
+   * Renders this component.
+   *
+   * @returns object JSX for this component.
+   */
+  render() {
+    const {
+      Fragment,
+    } = React;
+    const {
+      components: {
+        Notice,
+      },
+    } = wp;
+    const {
+      notifications,
+    } = this.state;
+
+    const visibleNotifications = notifications
+      .filter((notification) => true !== notification.dismissed);
+
+    return (
+      <Fragment>
+        {visibleNotifications.map((notification) => (
```

Comment:
Would this make sense as an unordered list?

Reply:
I thought about that. However, it's unlikely that there is going to be more than one alert in the vast majority of circumstances, and I'd have to undo a bunch of built-in list formatting to get the built-in styles for the Notice component to display properly.